runner/options.go

```diff
@@ -255,6 +253,7 @@ type Options struct {
 	OutputFilterCondition     string
 	OutputMatchCondition      string
 	OnResult                  OnResultCallback
+	indexFileData             *strings.Builder
```

Comment:
With large lists of input targets, the memory occupation will grow too much. We can open the output index file and sync-write to it, just like we do in the case of the output file (`-o` option)